pkg/download/config_creation/config_creation.go

```diff
@@ -0,0 +1,61 @@
+/*
+ * @license
+ * Copyright 2025 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package config_creation
+
+import (
+	"fmt"
+
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/config"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/config/coordinate"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/config/template"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/download/internal/templatetools"
+)
+
+// CreateConfig creates a config based on a given JSON payload and removes properties like the id (idPropertyKey), version and externalId
+func CreateConfig(projectName string, data []byte, configType config.Type, idPropertyKey string) (config.Config, error) {
```

Comment:
I would reduce the scope of this function to just removing unwanted entries. Than we could reuse it in every client. For example, the bucket client needs to delete:
```Go
	r.Delete(bucketName)
	r.Delete(status)
	r.Delete(version)
	r.Delete(updatable)
```
The signature should look something like: 
```Go
func Remove(data []byte, keys ... string) (json.JsonRaw, error)
```

What's your opinion @arthurpitman ?

Reply:
I adapted it to only return the id and the json payload. The config creation is then again up to the segments/slos

Reply:
I adapted it again. Now it's also possible to use it for download buckets. I haven't touched it yet, because this would change the error logs

regarding the delete suggestion: I would not vote for that as it would only transform it basically from vertical to horizontal with no real value

Reply:
Decision: Leave the duplicated code in for now and do some refactoring later on